public/types.ts

```diff
@@ -0,0 +1,9 @@
+import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';
```

Comment:
copyright

Reply:
will remove every changes under /public, to keep this PR pure for server side migration.